src/Apps/Legion/Components/UserDefinedEntityForm/UserDefinedEntityForm.view.tsx

```diff
@@ -0,0 +1,415 @@
+                            items={iconOptions}
+                            onChangeItem={(item) => {
+                                setSelectedIcon(item.id);
+                            }}
+                        />
+                        <ColorPicker
+                            label={t(LOC_KEYS.colorFieldLabel)}
+                            selectedItem={selectedColor}
+                            items={colorOptions}
+                            onChangeItem={(item) => {
+                                setSelectedColor(item.id);
+                            }}
+                        />
+                    </Stack>
+                )}
+
+                <CardboardComboBox
+                    label={t(LOC_KEYS.relationshipNameFieldLabel)}
+                    onSelectionChange={onNewEntityRelationshipChange}
+                    options={relationshipNameOptions}
+                    placeholder={t(LOC_KEYS.relationshipNameFieldPlaceholder)}
+                    selectedItem={newEntityRelationshipNameValue}
+                />
+            </>
+        );
+    }, [
+        colorOptions,
+        iconOptions,
+        isNewParentType,
+        newEntityRelationshipNameValue,
+        onNewEntityRelationshipChange,
+        onParentTypeChange,
+        parentEntityNameValue,
+        parentTypeOptions,
+        relationshipNameOptions,
+        selectedColor,
+        selectedIcon,
+        selectedParentType,
+        t
+    ]);
+
+    // RENDER EXISTING ENTITY FORM
+    const existingForm = useMemo(() => {
+        return (
+            <>
+                <Dropdown
+                    label={t(LOC_KEYS.parentEntityFieldLabel)}
+                    options={existingEntityOptions}
+                    onChange={(_ev, option) => {
+                        setSelectedExistingEntity(option);
+                    }}
+                    placeholder={t(LOC_KEYS.existingEntityFieldPlaceholder)}
+                    required
+                    selectedKey={selectedExistingEntity?.key}
+                />
+                <CardboardComboBox
+                    label={t(LOC_KEYS.relationshipNameFieldLabel)}
+                    onSelectionChange={onExistingEntityRelationshipChange}
+                    options={relationshipNameOptions}
+                    placeholder={t(LOC_KEYS.relationshipNameFieldPlaceholder)}
+                    selectedItem={existingEntityRelationshipNameValue}
+                />
+            </>
+        );
+    }, [
+        existingEntityOptions,
+        existingEntityRelationshipNameValue,
+        onExistingEntityRelationshipChange,
+        relationshipNameOptions,
+        selectedExistingEntity?.key,
+        t
+    ]);
+
+    logDebugConsole('debug', 'Render', parentTypeOptions, selectedParentType);
+
+    return (
+        <Stack className={classNames.root} tokens={{ childrenGap: 8 }}>
+            {formMode === 'New' ? newEntityForm : existingForm}
```

Comment:
Would we benefit from making each of these its own component?

Reply:
Was trying something new. This is the renderer and the main component fetches the data.

Reply:
I considered that. But there's enough overlap that I just thought it made sense to keep them together. You could probably split them and have it be fine but I didn't want to have to keep the props in sync between them and whatnot so this seemed simpler.

Reply:
There's also a very good chance this from will go away entirely so I figured it was better to just get this branch merged and if we still need it for H2 we will revamp it to fit the new models.